ambry-account/src/main/java/com/github/ambry/account/RouterStore.java

```diff
@@ -197,76 +215,65 @@ public ZNRecord update(ZNRecord znRecord) {
       } else {
         recordToUpdate = znRecord;
       }
-
-      String errorMessage = null;
-      List<String> accountBlobIDs = recordToUpdate.getListField(ACCOUNT_METADATA_BLOB_IDS_LIST_KEY);
+      // This is the version number for the new blob id.
       int newVersion = 1;
-      Map<String, String> accountMap = null;
-      if (accountBlobIDs != null && accountBlobIDs.size() != 0) {
-        // parse the json string list and get the blob id with the latest version
+      List<BlobIDAndVersion> blobIDAndVersions = new ArrayList<>();
```

Comment:
Maybe consider using a LinkedList if you are going to be removing from the start of the list, since that can be O(n) in an ArrayList.
Alternatively, you can sort in reverse order and remove from the end of the ArrayList

Reply:
good catch, I will use linkedlist instead.